apps/server/src/modules/tool/context-external-tool/service/tool-version-service.ts

```diff
@@ -0,0 +1,44 @@
+import { Injectable } from '@nestjs/common/decorators/core/injectable.decorator';
+import { Inject } from '@nestjs/common';
+import { ContextExternalToolValidationService } from './context-external-tool-validation.service';
+import { SchoolExternalToolValidationService } from '../../school-external-tool/service';
+import { IToolFeatures, ToolFeatures } from '../../tool-config';
+import { ExternalTool } from '../../external-tool/domain';
+import { SchoolExternalTool } from '../../school-external-tool/domain';
+import { ContextExternalTool } from '../domain';
+import { ToolConfigurationStatus } from '../../common/enum';
+import { CommonToolService } from '../../common/service';
+
+@Injectable()
+export class ToolVersionService {
+	constructor(
+		private readonly contextExternalToolValidationService: ContextExternalToolValidationService,
+		private readonly schoolExternalToolValidationService: SchoolExternalToolValidationService,
+		private readonly commonToolService: CommonToolService,
+		@Inject(ToolFeatures) private readonly toolFeatures: IToolFeatures
+	) {}
+
+	async determineToolConfigurationStatus(
+		externalTool: ExternalTool,
+		schoolExternalTool: SchoolExternalTool,
+		contextExternalTool: ContextExternalTool
+	): Promise<ToolConfigurationStatus> {
+		// TODO N21-1337 remove if statement, when feature flag is removed
+		if (this.toolFeatures.toolStatusWithoutVersions) {
+			try {
+				await this.schoolExternalToolValidationService.validate(schoolExternalTool);
+				await this.contextExternalToolValidationService.validate(contextExternalTool);
+				return ToolConfigurationStatus.LATEST;
+			} catch (err) {
```

Comment:
Should we debug log this?

Reply:
we could get the information what exactly failed to make the tool outdated. I don't think that is necessary as we will get the failed scope in the next iteration of this feature, which should suffice. Do you think there is added value for someone to know that there is a new required parameter instead of just the info, that the context scope has to be updated?